src/backend/blockchain/endpoints.md

```diff
@@ -0,0 +1,109 @@
+# Blockchain API
+
+## View the current contents of the chain
+
+> `GET` <http://localhost/blockchain/chain>
+
+Required JSON: `None`
+
+Example return:
+
+```json
+{
+  "length": 1,
+  "chain": [
+    {
+      "hash": "0000",
+      "index": 0,
+      "nonce": 0,
+      "prev_hash": "0",
+      "timestamp": "1614063326.48",
+      "transaction": { "amount": 0, "from_address": "", "to_address": "" }
+    }
+  ]
+}
+```
+
+## Send a proof to the blockchain for the current mining session
+
+> `POST` <http://localhost/blockchain/proof>
+
+Required JSON:
+
+```json
+{ "proof": string }
+```
+
+Example return:
+
+```json
+TBD
+```
+
+## Register a new Wallet on the blockchain
+
+> `POST` <http://localhost/blockchain/wallet/addWallet>
+
+Required JSON:
+
+```json
+{ "walletId": string }
+```
+
+Example return:
+
+```json
+200 { "success": true }
```

Comment:
Minor change: can you change the true to True to be compliment with python? or is it supposed to be a string saying "true"?

Reply:
In markdown, `true` will be interpreted as a primitive type and will be represented as such when the Markdown is generated. If I were to use `True` instead, it would look like plain text and not a type. This is because booleans in JSON are represented as `true`/`false`, unlike in Python where they are capitalized. When Python unmarshals the JSON, it will change `true` to `True`, as can be seen in `test_routes.py`